src/Microsoft.ML.Dnn/ImageClassificationTrainer.cs

```diff
@@ -705,18 +705,20 @@ public ImageProcessor(Session session, string jpegDataTensorName, string resizeI
 
             public Tensor ProcessImage(in VBuffer<byte> imageBuffer)
             {
-                using var imageTensor = EncodeByteAsString(imageBuffer);
-                try
+                using (var imageTensor = EncodeByteAsString(imageBuffer))
                 {
-                    return _imagePreprocessingRunner.AddInput(imageTensor, 0).Run()[0];
-                }
-                catch (TensorflowException e)
-                {
-                    //catch the exception for images of unknown format
-                    if (e.HResult == -2146233088 && e.Message.Contains("Expected image (JPEG, PNG, or GIF), got unknown format"))
-                        return null;
-                    else
-                        throw;
+                    try
+                    {
+                        return _imagePreprocessingRunner.AddInput(imageTensor, 0).Run()[0];
+                    }
+                    catch (TensorflowException e)
+                    {
+                        //catch the exception for images of unknown format
+                        if (e.HResult == -2146233088)
+                            return null;
```

Comment:
Is there a way to notify users which image fails? What occurs if all images don't load? 

Is there an enum that we can pull the "-2146233088" magic number from? Seems fragile. Something like `TensorflowExceptions.UnknownImageFormat`.

Reply:
Yep, TF prints to the console.

---
In reply to: [339946888](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339946888) [](ancestors = 339946888)

Reply:
> TF prints to the console.

That's not going to help users who are in a GUI application - like Visual Studio Model Builder.

Reply:
@eerhardt model builder receives logging from image classification api through callback that is attached to mlcontext.Log.

Reply:
And how does the "which image fails" information get through the callback if TF prints it to the console?

Where is the `TF prints to the console` code? Will that information flow the callback?